doc/microsite/src/main/tut/distage/distage-framework-docker.md

```diff
@@ -19,3 +20,581 @@ libraryDependencies += "io.7mind.izumi" %% "distage-framework-docker" % "$izumi.
+  val ports = Seq(9200, 9300)
+
+  override def config: Config = {
+    Config(
+      image = "docker.elastic.co/elasticsearch/elasticsearch:7.7.0",
+      ports = ports.map(Docker.DockerPort.TCP(_)),
+      env = Map("discovery.type" -> "single-node")
+    )
+  }
+}
+
+class ElasticSearchPlusPostgresModule[F[_]: TagK] extends ModuleDef {
+  make[PostgresDocker.Container].fromResource {
+    PostgresDocker.make[F]
+  }
+
+  make[ElasticSearchDocker.Container].fromResource {
+    ElasticSearchDocker.make[F].dependOnDocker(PostgresDocker)
+  }
+}
+```
+
+Another example of dependencies between containers is in the "Docker Container Networks" section later in this document.
+
+### Availability and Health Checks
+
+The `healthCheck` properties of
+@scaladoc[Docker.ContainerConfig](izumi.distage.docker.Docker$$ContainerConfig) configure the
+health checks. A container resource will not be provided if the health checks did not succeed at
+acquire time. These are used to determine if an existing container can be used and if starting a
+fresh container succeeded.
+
+By default, the `healthCheck` is a port check of the ports from the container config. There are
+several standard checks provided in
+@scaladoc[ContainerHealthCheck](izumi.distage.docker.healthcheck.ContainerHealthCheck$) that can be
+combined to cover many common cases.
+
+The @scaladoc[availablePorts](izumi.distage.docker.DockerContainer#availablePorts) property of
+the container resource are the mapped ports that passed the health check. This is a map
+from a `Docker.DockerPort` provided in the config to a host and port. For example:
+
+```scala
+val port = container.availablePorts.first(PostgresDocker.primaryPort)
+```
+
+would be a host and port mapped to the Postgres container's primary port that have passed the
+health check.
+
+### Usage in Integration Tests
+
+A common use case is using Docker containers to provide service implementations for integration test, such as using a PostgreSQL container for verifying an application that uses a PostgreSQL database.
+`distage` container resources are easy to integrate as providers.
+
+Consider the example application below. This application is written to depend on a
+[doobie](https://tpolecat.github.io/doobie/) `Transactor`, which is constructed from a
+`PostgresServerConfig`.
+
+```scala mdoc:silent
+import doobie._
+import doobie.implicits._
+import java.util.Date
+
+class PostgresExampleApp(xa: Transactor[IO]) {
+  def plusOne(a: Int): IO[Int] =
+    sql"select ${a} + 1".query[Int].unique.transact(xa)
+
+  val run: IO[Unit] = plusOne(1) flatMap { v => IO(println(s"1 + 1 = ${v}.")) }
+}
+
+// the postgres configuration used to construct the Transactor
+case class PostgresServerConfig(
+  host: String,
+  port: Int,
+  database: String,
+  username: String,
+  password: String
+)
+
+class TransactorFromConfigModule extends ModuleDef {
+  import scala.concurrent.ExecutionContext.Implicits.global
```

Comment:
optional nit, `global` may be imported non-implicitly since it's passed explicitly below
```suggestion
  import scala.concurrent.ExecutionContext.global
```